teos/watcher.py

```diff
@@ -673,11 +673,47 @@ def get_user_info(self, user_id):
             user_id (:obj:`str`): the id of the requested user.
 
         Returns:
-            :obj:`UserInfo <teos.gatekeeper.UserInfo> or :obj:`None`: The user data if found. :obj:`None` if not found, or
-            the ``user_id`` is invalid.
+            :obj:`UserInfo <teos.gatekeeper.UserInfo> or :obj:`None`: The user data if found. :obj:`None` if not found,
+            or the ``user_id`` is invalid.
         """
         return self.gatekeeper.registered_users.get(user_id)
 
+    def get_subscription_info(self, signature):
+        """
+        Gets information about a user's subscription.
+
+        Args:
+            signature (:obj:`str`): the signature of the request by the user.
+
+        Returns:
+            :obj:`UserInfo <teos.gatekeeper.UserInfo> or :obj:`None`: The user's subscription data if found. :obj:`None`
+            if not found, or the ``user_id`` is invalid.
+
+        Raises:
+            :obj:`SubscriptionExpired`: If the user subscription has expired.
+        """
+
+        message = "get subscription info".encode("utf-8")
+        user_id = self.gatekeeper.authenticate_user(message, signature)
+        if self.gatekeeper.has_subscription_expired(user_id):
+            raise SubscriptionExpired(
+                f"Your subscription expired at block {self.gatekeeper.registered_users[user_id].subscription_expiry}"
+            )
+
+        subscription_info = self.gatekeeper.registered_users.get(user_id)
+
+        locators = []
+
+        for appt_uuid in subscription_info.appointments:
+            if appt_uuid in self.appointments:
```

Comment:
Not exactly. You're returning all the data here, when only the locator was needed.

`self.appointments.get(appt_uuid).get("locator")` and `self.responder.trackers.get(appt_uuid).get("locator")` should do the trick.

Furthermore, I'd add an else with a log.debug recording that the locator for the uuid was not found. This should not happen, but if it does at least we'll have logs for it.